modules/images/webp-uploads/load.php

```diff
@@ -42,60 +46,55 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 		return $metadata;
 	}
 
-	$dirname     = pathinfo( $file, PATHINFO_DIRNAME );
-	$image_sizes = array();
-	if ( array_key_exists( 'sizes', $metadata ) && is_array( $metadata['sizes'] ) ) {
-		$image_sizes = $metadata['sizes'];
-	}
-
-	foreach ( wp_get_registered_image_subsizes() as $size_name => $properties ) {
-		// This image size does not exist on the defined sizes.
-		if ( ! isset( $image_sizes[ $size_name ] ) || ! is_array( $image_sizes[ $size_name ] ) ) {
+	$dirname = pathinfo( $file, PATHINFO_DIRNAME );
+	foreach ( $metadata['sizes'] as $size_name => $properties ) {
+		// This image size is not defined or not an array.
+		if ( ! isset( $metadata['sizes'][ $size_name ] ) || ! is_array( $metadata['sizes'][ $size_name ] ) ) {
 			continue;
 		}
 
-		$current_size = $image_sizes[ $size_name ];
-		$sources      = array();
-		if ( isset( $current_size['sources'] ) && is_array( $current_size['sources'] ) ) {
-			$sources = $current_size['sources'];
+		// Ensure a `sources` property exists on the existing size.
+		if ( empty( $metadata['sizes'][ $size_name ]['sources'] ) || ! is_array( $metadata['sizes'][ $size_name ]['sources'] ) ) {
+			$metadata['sizes'][ $size_name ]['sources'] = array();
 		}
```

Comment:
Throughout this entire loop, the code would be much easier to read if we always looked at the `$properties` variable which is the same as `$metadata['sizes'][ $size_name ]`. The only difference is that at the very end we have to reassign the whole variable like `$metadata['sizes'][ $size_name ] = $properties;` so that the data is also updated outside of this loop.

This feedback applies to the entire `foreach` loop, here specifically it would become
```suggestion
		if ( empty( $properties['sources'] ) || ! is_array( $properties['sources'] ) ) {
			$properties['sources'] = array();
		}
```

Reply:
Actually, we are doing 2 operations here read and write, while doing the read we can use properties but for write operations, we need to make sure the `metadata` is updated as it removes the need to create an auxiliary variable and we can call `wp_update_attachment_metadata` with the`metadata` variable instead of the `properties` one.

Reply:
I see there are some places that can be updated to make this simpler tho, so will update accordingly.